### PR TITLE
Added support for gateway sharding inside gateway and mdcb

### PR DESCRIPTION
**Pull request by andrei-tyk**
## Screenshots (if appropriate)
Pasted Graphic

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: components/tyk-gateway/values.yaml [hunk 1]
```diff
@@ -233,3 +239,7 @@ gateway:
   #    filename: test.txt
   #    mountPath: /opt/test
   mounts: []
+
+  # used to decide whether to send the results back directly to Tyk without a hybrid pump
+  # if you want to use a hybrid pump, please change "rpc" to ""
+  analyticsConfigType: "rpc"
```
Comment (caroltyk) [inline components/tyk-gateway/values.yaml:245]: I think default for gateway component should be ""
(Set empty for a Self-Managed installation or rpc for multi-cloud.)

File: components/tyk-pump/templates/deployment-pmp.yaml [hunk 2]
```diff
@@ -55,6 +55,8 @@ spec:
             value: "128"
          {{ if has "hybrid" .Values.pump.backend }}
           # Configure Hybrid Pump for Tyk Pump
+          # If you want to use hybrid pump with MDCB please also set
```
Comment (komalsukhani) [inline components/tyk-pump/templates/deployment-pmp.yaml:58]: It will be useful to move this comment to values file

Comment (andrei-tyk) [reply components/tyk-pump/templates/deployment-pmp.yaml]: Fixed

File: components/tyk-gateway/values.yaml (hunk 1 above)
Comment (andrei-tyk) [reply components/tyk-gateway/values.yaml]: Fixed

**Review by buraksekili — APPROVED** [review]

**Pull request merged**